### ISSUE TITLE
Getting empty worksheets in workbook created from datatables.

### ISSUE THREAD

**Issue by tysongibby** [ClosedXML/ClosedXML#1491]
## Read and complete the full issue template

Do not randomly delete sections. They are here for a reason.

**Do you want to request a *feature* or report a *bug*?**
- [x] Bug
- [ ] Feature
- [ ] Question

**Did you test against the latest CI build?**
- [x] Yes
- [ ] No

If you answered `No`, please test with [the latest development build](https://ci.appveyor.com/project/ClosedXML/ClosedXML/branch/develop/artifacts) first.

**Version of ClosedXML**
0.95.3

**What is the current behavior?**
workbook is created with worksheets using fully populated datatables. Worksheets are created with the correct number of columns and rows, but no data from the datatables is being populated into the worksheet cells. Columns are only labled as Column1, Column2, Column3 etc... Worksheets do have the correct names from the datatable names. While debugging I an view the datatable data from the Visual Studio visualizer. The table data looks well-formed.
(https://github.com/ClosedXML/ClosedXML/files/5063907/Daily_Reports_12-08-2020_09-32.xlsx)

**What is the expected behavior or new feature?**
data from data tables to be populated in the worksheets of workbook.

**Is this a regression from the previous version?**

Regressions get higher priority. Test against the latest build of the previous minor version. For example, if you experience a problem on v0.95.3, check whether it the problem occurred in v0.94.2 too. 

Problem also occurs in v0.95.2.

## Reproducibility
**This is an important section. Read it carefully. Failure to do so will cause a 'RTFM' comment.**

Without a code sample, it is unlikely that your issue will get attention. Don't be lazy. Do the effort and assist the developers to reproduce your problem. Code samples should be [minimal complete and verifiable](https://stackoverflow.com/help/mcve). Sample spreadsheets should be attached whenever applicable. Remove sensitive information.

Unfortunately I am not able to include actual data from the datatable..

```c#
public void Main()
{
public XLWorkbook ExportExcelFiles(string checkedReportsString)
        {           
            using (var wb = new XLWorkbook())
            {
                //find each report id in the url parameter sting and create and excel workbook with a worksheet for each one
                string reportId = null;
                foreach (char c in checkedReportsString)
                {
                    
                    //find the next report id in url parameter string and create an excel worksheet from it
                    if (c != char.Parse("~"))
                    {
                        reportId = reportId + c;
                        continue;
                    }

                    else if (c == char.Parse("~"))
                    {
                        //create datatable of the current report
                        string reportName = GetReportTableName(reportId);
                        DataTable reportDataTable = GetTableData(reportId, reportName);
                        
                        //add datatable to workbook                                                
                        if (reportDataTable != null && reportDataTable.Rows.Count > 0)
                        {
                            reportDataTable.TableName = $"{GetReportTableName(reportId).Substring(7)}_{GetReportDateStringWithTime(reportId)}";                            
                            var ws = wb.Worksheets.Add(reportDataTable);                            
                        }
                        reportId = null;
                    }
                }
                if (wb != null) return wb;
                else throw new Exception("No Data to Report");

                
            }             
        }
}
```

- [x ] I attached a sample spreadsheet.  (You can drag files on to this issue)

[Daily_Reports_12-08-2020_09-32.xlsx]
[testWorkbook.xlsx](https://github.com/ClosedXML/ClosedXML/files/5052853/testWorkbook.xlsx)

**Comment by igitur** [comment]
> Unfortunately I am not able to include actual data from the datatable..

Not much I can do then.

**Comment by tysongibby** [comment]
This function does work in ClosedXML version 0.94.2

**Comment by igitur** [comment]
Ok, give me a reproducible test case in line with the requirements ( https://stackoverflow.com/help/mcve ) and we can look into it.

**Comment by tysongibby** [comment]
Hi Igitur,  for the DataTable we are using, it is the result set of a select statement from an Oracle Database that is loaded.  I do not mean to be difficult, but I will not be able to provide any information than this and the other information I provided previously. I am sorry. I understand if you are unable to address it because this is not enough to work with. This is an open source project to which you graciously dedicate your time. I appreciate you and I am grateful the older version  works. Thank you.

**Comment by vbjay** [comment]
You fail to understand how to do an mvce.

Forget the database. Write code to generate a datatable with sample info
that mimics your result structure.  Create a console app that generates the
datatable and then does the stuff you want with closedxml.

**Comment by igitur** [comment]
@tysongibby Can you not recreate your Oracle dataset based on dummy values, but still reproducing the error?

@vbjay I've made very snarky in the past myself and I regret that, but I'm trying to be a better person now. Let's try to be more welcoming. Because we definitely need more help with the project here. Don't want to scare people away.

**Comment by vbjay** [comment]
Sure thing.

**Comment by tysongibby** [comment]
Hi vbjay, I appreciate your feedback and I do agree with igitur. You can accomplish the same exchange of information in a more professional and polite way. Not only does it encourage more participation, many employers look us up on GitHub and make decisions on our applications based on our work and our conduct. 

I support what you are doing on this project and appreciate the years of labor that has been put into something of this size and function. It is awesome in the truest sense of the word.  

To your point, I have done what you are suggesting and it works without issue. When I bring the data over from the Oracle Database, it creates all of the columns (the exact number) and all of the rows (the exact number) with filtering enabled on each column and alternating blue and white row formatting, but there is no data in any of the cell.  I would like to include more information than this, but it is not possible. If you have a question I will be glad to answer it if I can. If this makes it impossible for the issue to be resolved, I understand.

**Comment by Pankraty** [comment]
I am 90% sure the reason is that you return a disposed workbook out of the method. Try remove `using` and let the caller dispose the workbook after it is saved.

**Comment by igitur** [comment]
@tysongibby Please triple check whether your code sample is indeed the *minimal* code to reproduce the issue. i.e. if @Pankraty  is right with his claim about the object being disposed, then the `DataTable` is totally irrelevant.

**Comment by tysongibby** [comment]
@Pankraty @igitur  To recreate and test the issue I created an app that generated a datatable as @vbjay has suggested. This worked without error and the excel workbook was created as expected with all columns, rows, and data and no formatting for styling. Then I tried using the same code but used the DataTable generated from the Oracle SQL select result instead. This still ended with a workbook that had all columns (with incorrect names.. ie.. column1, column2, column3, etc... instead) and all the rows, but no data in any of the cells (see attached spreadsheet from first submission of this issue). Also the columns had filtering turned on and the rows had styling that alternated light blue and white in color (see attached spreadsheet from first submission of this issue).   

I am not going to pretend to understand OpenXml. I peaked at the code you guys have created and it would take me some time to comprehend what is happening there. I admire what you have created. 

To an ignorant outsider, it seems that that at some point during the parsing of the DataTable a part of the data is getting confused as formatting and styling information and therefore not allowing the cell and column data to be properly parsed.

**Comment by igitur** [comment]
Change your code to this and try again:
```c#
public XLWorkbook ExportExcelFiles(string checkedReportsString)
        {           
            var wb = new XLWorkbook();
            
                //find each report id in the url parameter sting and create and excel workbook with a worksheet for each one
                string reportId = null;
                foreach (char c in checkedReportsString)
                {
                    
                    //find the next report id in url parameter string and create an excel worksheet from it
                    if (c != char.Parse("~"))
                    {
                        reportId = reportId + c;
                        continue;
                    }

                    else if (c == char.Parse("~"))
                    {
                        //create datatable of the current report
                        string reportName = GetReportTableName(reportId);
                        DataTable reportDataTable = GetTableData(reportId, reportName);
                        
                        //add datatable to workbook                                                
                        if (reportDataTable != null && reportDataTable.Rows.Count > 0)
                        {
                            reportDataTable.TableName = $"{GetReportTableName(reportId).Substring(7)}_{GetReportDateStringWithTime(reportId)}";                            
                            var ws = wb.Worksheets.Add(reportDataTable);                            
                        }
                        reportId = null;
                    }
                }
                if (wb != null) return wb;
                else throw new Exception("No Data to Report");

                
            }             
        
```

**Comment by tysongibby** [comment]
@Pankraty @vbjay @igitur Please excuse the late response.  I removed the using statement as Pankraty suggested and brought my ClosedXML version back up to the most current release and the issue of the data missing from the cells has gone away. The pre-formatted styling is still in place. Thank you for your help.

**Comment by vbjay** [comment]
Great. Yes. You can't hand off anything that is in a using block. It will
be disposed.

void test()
{
Something TMP = GetSomething();
}

Something GetSomething()
{
using (Something x = blaaa())
{
return x;
}
}

See the problem?

**Comment by vbjay** [comment]
Great. Yes. You can't hand off anything that is in a using block. It will
be disposed.
```c#
void test()
{
Something TMP = GetSomething();
}

Something GetSomething()
{
    using (Something x = blaaa())
    {
        return x;
    }
}

See the problem?

GitHub email doesn't play nice with code.

**Comment by tysongibby** [comment]
Yes, Thanks for the help in troubleshooting the issue.

**Comment by igitur** [comment]
Fixed in #1499
